convert2rhel/checks.py

```diff
@@ -75,81 +93,79 @@ def perform_pre_ponr_checks():
 
 
 def check_convert2rhel_latest():
-    """Make sure that we are running the latest version of convert2rhel"""
-    gpg_path = os.path.join(utils.DATA_DIR, "gpg-keys", "RPM-GPG-KEY-redhat-release")
-    ssl_cert_path = os.path.join(utils.DATA_DIR, "redhat-uep.pem")
-    repo_content = (
-        "[convert2rhel]\n"
-        "name=Convert2RHEL Repository\n"
-        "baseurl=https://cdn.redhat.com/content/public/convert2rhel/$releasever/x86_64/os/\n"
-        "gpgcheck=1\n"
-        "enabled=1\n"
-        "sslcacert=%s\n"
-        "gpgkey=file://%s\n" % (ssl_cert_path, gpg_path)
-    )
+    """Make sure that we are running the latest downstream version of convert2rhel"""
+    logger.task("Prepare: Checking if this is the latest version of Convert2RHEL")
+
+    if not system_info.has_internet_access:
+        logger.warning("Skipping the check because no internet connection has been detected.")
+        return
 
     cmd = [
         "repoquery",
+        "--disablerepo=*",
         "--releasever=%s" % system_info.version.major,
+        "--repofrom=convert2RHELrepo,%s" % CDN_URL,
+        "convert2rhel",
```

Comment:
@Andrew-ang9, I'm sure you know that this main part of the whole feature, running the repoquery command, doesn't work. First, you're disabling all repos, including the one you're defining (convert2RHELrepo). Second, you're not using the certificate so the repoquery request is refused. Do you have at least a working repoquery command at hand that you are about to implement here?

Reply:
@bocekm I worked with Andrew on this.... We tested this on a Centos8 box and found that `--repofrom` ignores the  `--disablerepo`.  That did surprise me.  It occurs to me that we should have tested this on other versions as well, though.  I'm now wondering if `--disablerepo=*` takes precedence on RHEL7, where the backend is yum-utils instead of all being a part of dnf. (Although testing on the matrix of distros that we support is one of the reasons we have CI right?  to catch issues from differences in the various distros?)

I'm trying to reason about the certificate in my head..... If the user has installed convert2rhel using the procedure we publicize here: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/converting_from_an_rpm-based_linux_distribution_to_rhel/index#proc_preparing-for-a-rhel-conversion_converting-from-a-linux-distribution-to-rhel  then they will have installed the certificate with `curl --create-dirs -o /etc/rhsm/ca/redhat-uep.pem https://ftp.redhat.com/redhat/convert2rhel/redhat-uep.pem`.  Ah, but we're creating our own repository so the certificate's location on disk won't be known.  Right... So we need something like `--setopt=sslcacert=PATH/TO/redhat-uep.pem`  (In testing, I lead Andrew astray... I thought that when installed a s documented, the certificate would be available in the system, so I had him add it on the system we were testing on [using update-ca-trust].  But we aren't telling the user to install the certificate into ca-trust, just to drop into a file on the filesystem, and then that specific file is referenced in the convert2rhel.repo file....)

And now I've checked RHEL7.... indeed, on RHEL7, we need `--disablerepo=* --enablerepo=convert2RHELrepo`.  And on RHEL7, `--setopt=sslcacert=redhat-uep.pem` does not work (but it does work on RHEL8)....  Going to have to see if there's a way to fix that on the command line or if we'll have to generate an actual convert2rhel.repo file.

Reply:
Yeah, I was trying it on RHEL 7 and `--disablerepo=* --enablerepo=convert2RHELrepo` did the trick regarding the disabled convert2RHELrepo.

Regarding the cert, I was trying `--setopt=convert2RHELrepo.sslcacert=/usr/share/convert2rhel/redhat-uep.pem`, described in yum man, but that didn't work either. I was thinking that we may need to get back to the original solution - creating the .repo file.

By the way, I was surprised that `--repofrom` works when the man page mentions just `--repofrompath`.

Reply:
Since repoquery on RHEL7's documentation implies that `--setopt=sslcacert=FILE` should work but it doesn't, I had to be creative in looking for a suitable alternative.  It looks like we can go back to creating a repo file that points to the convert2rhel repo, setting the path to the ssl certificate in there, and then telling repoquery to use that repofile by pointing it at the directory that holds the file.  For instance:

`repoquery --releasever=8 convert2rhel --setopt=reposdir=convert2rhel/data/version-independent/yum.repos.d`

That's a bit of a hack, but since RHEL7 doesn't pass the `sslcacert` setting from the commandline to curl for `--repofrom` and there doesn't appear to be a way to pass a repo file to repoquery, it might be the best we can do.

Reply:
This should be done now.  repo is back to being set via  a file.